Gems/LyShine/Code/Source/UiCanvasComponent.cpp

```diff
@@ -1874,10 +1874,35 @@ AZ::RHI::AttachmentId UiCanvasComponent::UseRenderTarget(const AZ::Name& renderT
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 AZ::RHI::AttachmentId UiCanvasComponent::UseRenderTargetAsset(const AZ::Data::Asset<AZ::RPI::AttachmentImageAsset>& attachmentImageAsset)
 {
+    // Check that the attachment image asset's bind flags are compatible
+    // - Supports use as a color attachment on a scope
+    // - Supports read access through a ShaderResourceGroup
+    const AZ::RHI::ImageDescriptor& imageDescriptor = attachmentImageAsset->GetImageDescriptor();
+    bool isCompatible = AZ::RHI::CheckBitsAll(imageDescriptor.m_bindFlags,
+        AZ::RHI::ImageBindFlags::Color | AZ::RHI::ImageBindFlags::ShaderRead);
+    if (!isCompatible)
+    {
+        AZ_Error("UI", false, "Attachment image asset: %s is not compatible for rendering a UI Canvas. Please "
+            "ensure that its BindFlags property includes Color and ShaderRead.", attachmentImageAsset.GetHint().c_str());
+        return AZ::RHI::AttachmentId();
+    }
+
+    // Check that the attachment image asset's pixel format is compatible
+    isCompatible = (imageDescriptor.m_format == AZ::RHI::Format::R8G8B8A8_UNORM)
+        || (imageDescriptor.m_format == AZ::RHI::Format::R8G8B8A8_UNORM_SRGB);
+    if (!isCompatible)
+    {
+        AZ_Error("UI", false, "Attachment image asset: %s has an invalid pixel format for rendering a "
+            "UI Canvas. Please ensure that it contains a Format property with a valid value of "
+            "19 (R8G8B8A8_UNORM) or 20 (R8G8B8A8_UNORM_SRGB).", attachmentImageAsset.GetHint().c_str());
+        return AZ::RHI::AttachmentId();
+    }
+
     auto attachmentImage = AZ::RPI::AttachmentImage::FindOrCreate(attachmentImageAsset);
     if (!attachmentImage)
     {
-        AZ_Warning("UI", false, "Failed to find or create render target");
+        AZ_Error("UI", false, "Failed to find or create render target from asset: %s",
+            attachmentImageAsset.GetHint().c_str());
```

Comment:
If there is no attachmentImage, how can it do 'GetHint()'? I think that'll give a nullptr exception right?

Reply:
GetHint is called on attachmentImageAsset though, can this function even fail if attachmentImageAsset is valid (or can it be invalid as well, then might worth checking that in the beginning of the function)?